.github/workflows/deploy-to-production.yaml

```diff
@@ -0,0 +1,218 @@
+
+      # Debugging step that lists the Hugo-generated images
+      # directory *after* the imache cache restore step,
+      # to make sure that the restore happened successfully
+    - name: List contents of images dir
+      continue-on-error: true
+      run: ls -al ${{ env.hugo_image_cache_path }}
+
+    - name: Use Node.js
+      uses: actions/setup-node@v3
+      with:
+        node-version: 14
+
+    - name: Install dependencies (Node)
+      working-directory: ./docs-repo
+      run: npm ci
+
+    - name: Build Hugo
+      env:
+        HUGOxPARAMSxSEARCH_CONFIG2xAPP_ID: ${{ vars.ALGOLIA_APP_ID }}
+        HUGOxPARAMSxSEARCH_CONFIG2xAPI_KEY: ${{ vars.ALGOLIA_SEARCH_KEY }}
+      working-directory: ./docs-repo
+      run: |
+        hugo config
+        hugo -b "${{ vars.DOCS_WEBSITE_URL }}" --gc --minify -d public
+
+      # Debugging step that lists the Hugo-rendered files
+      # to make sure the site was built
+    - name: List rendered files
+      working-directory: ./docs-repo
+      run: |
+        sudo apt install -y tree
+        tree -L 1 public
+
+    - name: Set up SSH agent (docs webserver key)
+      uses: webfactory/ssh-agent@v0.7.0
+      with:
+        ssh-private-key: ${{ secrets.DOCS_WEBSITE_RSYNC_PRIVATE_KEY }}
+
+    - name: Rsync site to webserver
+      run: |
+        echo "${{ secrets.DOCS_WEBSITE_RSYNC_HOST_VERIFICATION_1 }}" > rsync_known_hosts
+        echo "${{ secrets.DOCS_WEBSITE_RSYNC_HOST_VERIFICATION_2 }}" >> rsync_known_hosts
+
+        rsync_to_destination() {
+          # Sync these assets without the --delete flag. These
+          # assets' filenames are fingerprinted, so they will not
+          # conflict with old versions that remain on the server.
+          rsync -r -e "ssh -o UserKnownHostsFile=rsync_known_hosts" \
+            docs-repo/public/css \
+            docs-repo/public/images \
+            docs-repo/public/js \
+            docs-repo/public/jslibs \
+            docs-repo/public/linode \
+            ${{ secrets.DOCS_WEBSITE_USER }}@$1:${{ secrets.DOCS_WEBSITE_RSYNC_DIR }}
+
+          # Sync everything else, and delete files in the destination
+          # rsync dir that do not exist in the source docs-repo/public/
+          # folder
+          rsync -r -e "ssh -o UserKnownHostsFile=rsync_known_hosts" \
+            --exclude="/css/" \
+            --exclude="/images/" \
+            --exclude="/js/" \
+            --exclude="/jslibs/" \
+            --exclude="/linode/" \
+            --delete \
+            docs-repo/public/ \
+            ${{ secrets.DOCS_WEBSITE_USER }}@$1:${{ secrets.DOCS_WEBSITE_RSYNC_DIR }}
+        }
+
+        rsync_to_destination "${{ secrets.DOCS_WEBSITE_RSYNC_DESTINATION_1 }}"
+        rsync_to_destination "${{ secrets.DOCS_WEBSITE_RSYNC_DESTINATION_2 }}"
+
+        rm rsync_known_hosts
+
+    - name: Update Algolia
+      working-directory: ./linode-docs-theme-repo
+      env:
+        ALGOLIA_APP_ID: ${{ vars.ALGOLIA_APP_ID }}
```

Comment:
Note that I have created a separate testing Algolia app (also separate from the sandbox) for this production-update workflow so that we don't interfere with the actual production Algolia app for now.

Reply:
This separate Algolia app is now failing to be updated in the production workflow, because it hit its usage limits for the free tier it's on. But, it was working for a few runs, so this isn't concerning to me right now